src/main/java/io/pravega/perf/PravegaPerfTest.java

```diff
@@ -289,23 +242,180 @@ private static void parseCmdLine(String[] args) throws ParseException {
+                }
+
+                writeNread = consumerCount > 0;
+
+                if (writeNread) {
+                    produceStats = null;
+                } else {
+                    produceStats = new PerfStats("Writing", REPORTINGINTERVAL, messageSize, writeFile);
+                }
+                eventsPerProducer = (events + producerCount - 1) / producerCount;
+                if (throughput == 0 && runtimeSec > 0) {
+                    eventsPerSec = events / producerCount;
+                } else if (throughput > 0) {
+                    eventsPerSec = (int) (((throughput * 1024 * 1024) / messageSize) / producerCount);
+                } else {
+                    eventsPerSec = 0;
+                }
+            } else {
+                produceStats = null;
+                eventsPerProducer = 0;
+                eventsPerSec = 0;
+                writeNread = false;
+            }
+
+            if (consumerCount > 0) {
+                String action;
+                if (writeNread) {
+                    action = "Write/Reading";
+                } else {
+                    action = "Reading";
+                }
+                consumeStats = new PerfStats(action, REPORTINGINTERVAL, messageSize, readFile);
+                eventsPerConsumer = events / consumerCount;
+            } else {
+                consumeStats = null;
+                eventsPerConsumer = 0;
+            }
+
+        }
+
+
+        private void start(long startTime) throws IOException {
+            if (produceStats != null) {
+                produceStats.start(startTime);
+            }
+            if (consumeStats != null) {
+                consumeStats.start(startTime);
+            }
+        }
+
+        private void shutdown(long endTime) {
+            try {
+                if (produceStats != null) {
+                    produceStats.shutdown(endTime);
+                }
+                if (consumeStats != null) {
+                    consumeStats.shutdown(endTime);
+                }
+            } catch (ExecutionException | InterruptedException ex) {
+                ex.printStackTrace();
             }
         }
+
+        public abstract List<Callable<Void>> getProducers();
+
+        public abstract List<Callable<Void>> getConsumers() throws URISyntaxException;
+
     }
 
-    private static class StartLocalService {
-        static final int PORT = 9090;
-        static final String SCOPE = "Scope";
-        static final String STREAM_NAME = "aaj";
+    static private class PravegaTest extends Test {
+        final PravegaStreamHandler streamHandle;
+        final ClientFactory factory;
+
+        PravegaTest(long startTime, CommandLine commandline) throws
+                IllegalArgumentException, URISyntaxException, InterruptedException, Exception {
+            super(startTime, commandline);
+            final ScheduledExecutorService bgExecutor = Executors.newScheduledThreadPool(10);
```

Comment:
Is 10 chosen for a reason?

Reply:
Thanks Tom.. No , it its existing since quite some time.. please let me know if you have any preference value for this.